internal/diagnostics/noderesourceinfo.go

```diff
@@ -0,0 +1,251 @@
+package diagnostics
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/kyma-project/cli.v3/internal/kube"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+)
+
+type MachineInfo struct {
+	Name             string
```

Comment:
I would suggest adding additional `yaml:"<name>"` tags for all fields to print yaml in `camelCase` format